remotes/mount.go

```diff
@@ -151,16 +151,25 @@ func (r *remoteReaderAt) ReadAt(p []byte, off int64) (int, error) {
 }
 
 type descriptorAccumulator struct {
-	descriptors []ocischemav1.Descriptor
+	descriptors     []ocischemav1.Descriptor
+	childrenHandler images.HandlerFunc
+	resolver        docker.ResolverBlobMounter
+	targetRepo      string
 }
 
 func (a *descriptorAccumulator) Handle(ctx context.Context, desc ocischemav1.Descriptor) ([]ocischemav1.Descriptor, error) {
+	_, _, err := a.resolver.Resolve(ctx, fmt.Sprintf("%s@%s", a.targetRepo, desc.Digest))
+	if err == nil {
+		fmt.Fprintf(os.Stderr, "Ignoring descriptor %s, as it is already present in target repository\n", desc.Digest)
+		// already exsits, do not process
+		return nil, nil
+	}
 	descs := make([]ocischemav1.Descriptor, len(a.descriptors)+1)
 	descs[0] = desc
 	for i, d := range a.descriptors {
 		descs[i+1] = d
 	}
```

Comment:
Not related to this change, but isn't this open coding a variant of `append`?

Reply:
It is actually a Prepend

Reply:
Right, that is more usually written as:
```go
a.descriptors = append([]ocischemav1.Descriptor{desc}, a.descriptors...)
```

Reply:
Getting even further off-topic: can't this whole `accumulator` thing be avoided by chaining `handler` (a `copier` here) onto the calling `images.Handlers` (perhaps after the `sourceFetcher`, not sure if there is an ordering constraint here)?

Collecting the descriptors from the walk in order to process them seems like an antipattern compared with dealing with them during the walk, but perhaps there is a good reason for it?

Reply:
Yes, you got it right, the order matters :(
+ I want to have a good UI at the end where we see some kind of a progress bar.